src/components/application_manager/src/commands/mobile/set_global_properties_request.cc

```diff
@@ -298,6 +298,14 @@ bool SetGlobalPropertiesRequest::PrepareResponseParameters(
                       ui_response_info_);
     return result;
   }
+  if (ui_properties_info.is_ok && tts_properties_info.is_unsupported_resource) {
```

Comment:
@pvvasilev I propose you to move all this if-s between 290-308 lines to overriden `PrepareResultForMobileResponse/PrepareResultCodeForResponse` functions as it's done in pull requests #1575, #1567 and #1562 for other RPCs with similar defect.